core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java

```diff
@@ -275,6 +291,10 @@ private ConnectivityState getAggregatedState() {
       return checkNotNull(subchannel.getAttributes().get(STATE_INFO), "STATE_INFO");
     }
 
+    static PickResult getSubchannelPickResult(Subchannel subchannel) {
```

Comment:
I did it like this for encapsulation - in the current structure `Picker` is a peer rather than child of `RoundRobinLoadBalancer`, and does not have knowledge of the Subchannel attributes.

But this will be moved to a different PR regardless per the earlier comment.